File: macos/QMK Toolbox/USB/Bootloader/USBAspDevice.m
#import "USBAspDevice.h"

@implementation USBAspDevice

- (id)initWithUSBDevice:(USBDevice *)usbDevice {
    if (self = [super initWithUSBDevice:usbDevice]) {
        self.name = @"USBasp";
        self.type = BootloaderTypeUsbAsp;
        self.eepromFlashable = YES;
    }
    return self;
}

-(void)flashWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"avrdude" withArgs:@[@"-p", mcu, @"-c", @"usbasp", @"-U", [NSString stringWithFormat:@"flash:w:%@:i", file], @"-C", @"avrdude.conf"]];
}

-(void)flashEEPROMWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"avrdude" withArgs:@[@"-p", mcu, @"-c", @"usbasp", @"-U", [NSString stringWithFormat:@"eeprom:w:%@:i", file], @"-C", @"avrdude.conf"]];
}

@end

File: macos/QMK Toolbox/USB/Bootloader/USBAspDevice.swift
import Foundation

class USBAspDevice: BootloaderDevice {
    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "USBasp"
        type = .usbAsp
        eepromFlashable = true
    }

    override func flash(_ mcu: String, file: String) {
        runProcess("avrdude", args: ["-p", mcu, "-c", "usbasp", "-U", "flash:w:\(file):i", "-C", "avrdude.conf"])
    }

    override func flashEEPROM(_ mcu: String, file: String) {
        runProcess("avrdude", args: ["-p", mcu, "-c", "usbasp", "-U", "eeprom:w:\(file):i", "-C", "avrdude.conf"])
    }
}
